test/functional/ui/inccommand_spec.lua

```diff
@@ -1354,6 +1354,12 @@ describe("inccommand=nosplit", function()
       :echo 'foo'^         |
     ]])
   end)
+
+  it("only executes the first command in preview", function()
+    insert(default_text)
+    feed(':%s/two/three/g|q!')
```

Comment:
This could be a bit racy, I would suggest adding a screen tests afterwards (this will show that `q!` has been received by nvim)

Reply:
Done